File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        ArrayList<Integer> ans = new ArrayList<>();
        ListNode temp = head;
        while(temp!=null){
            ans.add(temp.val);
            temp=temp.next;
        }

        int n = ans.size();
        int i=0;
        int j=n-1;
        while(i<j){
            if(ans.get(i)!=ans.get(j)) return false;
            else if(ans.get(i)==ans.get(j)){
                i++;
                j--;
            }
        }
        return true;
    }
}

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: bool
        """
        ans = []
        slow = head
        while slow != None:
            ans.append(slow.val)
            slow = slow.next
        return ans == ans[::-1]
            
        